src/libraries/Microsoft.PowerFx.Interpreter/Functions/IRandomService.cs

```diff
@@ -31,12 +39,25 @@ internal class DefaultRandomService : IRandomService
         [ThreadSafeProtectedByLock(nameof(_randomizerLock))]
         private static readonly Random _random = new Random();
 
+        private static readonly object _randomizerDecimalLock = new object();
+
+        [ThreadSafeProtectedByLock(nameof(_randomizerDecimalLock))]
+        private static readonly Random _randomDecimal = new Random();
```

Comment:
Do we need another Random object? We should be able to use the same as the one used to return double values.

Reply:
That should not be a problem - those methods are very small and should not have locking problems.

Reply:
Updated here